### PR TITLE
Refactor `votes` route to return 400 when a voter has not cast a vote

### PR DESCRIPTION
**Pull request by keefertaylor**
This is in line with behavior from the existing LCD API that ships in Cosmos SDK. 

I'm leaning fairly heavily on pattern matching and some help from ChatGPT to supplement my nascent rust knowledge, so there could be a better way. Feel free to drop this PR, or suggest changes if you know how to do this more elegantly.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -422,73 +423,85 @@ async fn proposals(args: &State<Args>) -> Value {
...
+                      "code": 3,
+                      "message": format!("voter: {} not found for proposal: {}", voter, proposal_id),
+                      "details": []
+                  })))
+              },
+              Some(option) => {
+                  let vote = match option.vote {
+                      3 => "VOTE_OPTION_NO",
+                      2 => "VOTE_OPTION_YES",
+                      1 => "VOTE_OPTION_YES",
```
Comment (freak12techno) [inline src/main.rs]: VOTE_OPTION_ABSTAIN?

**Comment by freak12techno** [comment]
@keefertaylor do you mind checking my comment? want to merge it

**Comment by keefertaylor** [comment]
Apologies for losing track of this, it's been a busy week. 

I fixed the abstain option. You can double check these values [here](https://github.com/penumbra-zone/penumbra/blob/08f3b2b0e97509169601f35b22160500932162e8/proto/penumbra/penumbra/core/component/governance/v1/governance.proto#L153).

**Comment by keefertaylor** [comment]
Also want to cal out that when I opened the PR I had this returning HTTP 404s for missing votes, but it seems the existing LCD impls in the Cosmos SDK actually return 400s.

That change got made in https://github.com/freak12techno/penumbra-lcd/pull/3/commits/82af838f574f983a7a3c3c744ea06bf654c2bb07.

**Review by freak12techno — APPROVED** [review]

**Comment by freak12techno** [comment]
@keefertaylor appreciate that, thanks!

**Pull request merged**